ark_sdk_python/services/pcloud/platforms/ark_pcloud_platforms_service.py

```diff
@@ -111,16 +112,21 @@ def platform(self, get_platform: ArkPCloudGetPlatform) -> ArkPCloudPlatform:
             ArkServiceException: _description_
 
         Returns:
-            ArkPCloudPlatform: _description_
+            Union[ArkPCloudPlatform, ArkPCloudPlatformDetails]: Platform details in appropriate format
         """
         self._logger.info(f'Retrieving platform [{get_platform.platform_id}]')
         resp: Response = self._client.get(PLATFORM_URL.format(platform_id=get_platform.platform_id))
         if resp.status_code == HTTPStatus.OK:
             try:
-                return ArkPCloudPlatform.model_validate(resp.json())
-            except (ValidationError, JSONDecodeError) as ex:
-                self._logger.exception(f'Failed to parse platform response [{str(ex)}] - [{resp.text}]')
-                raise ArkServiceException(f'Failed to parse platform response [{str(ex)}]') from ex
+                # Try the correct model for Details API first
+                return ArkPCloudPlatformDetails.model_validate(resp.json())
```

Comment:
For back compatability, i would suggest to swap things around, plat first and then this